firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -203,6 +205,21 @@ extension BrowserViewController: WKUIDelegate {
         completionHandler(contextMenuConfiguration(for: url, webView: webView))
     }
 
+    func webView(_ webView: WKWebView,
+                 requestMediaCapturePermissionFor origin: WKSecurityOrigin,
+                 initiatedByFrame frame: WKFrameInfo,
+                 type: WKMediaCaptureType,
+                 decisionHandler: @escaping (WKPermissionDecision) -> Void) {
+        // If the tab isn't the selected one or we're on the homepage, do not show the media capture prompt
+        guard tabManager.selectedTab?.webView == webView, !contentContainer.hasLegacyHomepage else {
```

Comment:
do we also need a check for the new homepage or private mode?

Reply:
Hello, yes good catch. This was only moved in this PR, but for sure the media permission cannot show on any type of homepages. I guess this can be changed in another PR

Reply:
I can open another PR to add the check for a new homepage or private mode. I will leave it out of this PR in case we want to revert these changes

Reply:
works for me! thank you~